tests/test_some_of.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import random
+import unittest
+
+from monai.transforms import Transform
+from monai.transforms.compose import SomeOf
+from monai.utils import set_determinism
+
+
+class A(Transform):
+    def __call__(self, x):
+        return 2 * x
+
+
+class B(Transform):
+    def __call__(self, x):
+        return 3 * x
+
+
+class C(Transform):
+    def __call__(self, x):
+        return 5 * x
+
+
+set_determinism(seed=0)
```

Comment:
thanks, please use this approach to avoid changing the global random states: https://github.com/Project-MONAI/MONAI/blob/a8302eca3fc31554d4bd30fae84150766ce13472/tests/test_grid_dataset.py#L32-L37

Reply:
Good spotting. Made the change.